### DISCUSSION TITLE
Using glass effect in the hubsroom, using spoke or blender

### DISCUSSION THREAD

**Discussion by Thitle** [mozilla/hubs#2727] [General Questions & Discussion] [answered]
Hi guys, I have a doubt, I have a project under development and I would like to use glass texture. Is there such a possibility? appreciate!

**Comment by emclaren** [comment]
Hi @Thitle - you can have a 3D model that has transparency like glass. For example, a few of the pieces in the architecture kit in Spoke are transparent windows. 

Note that these transparencies are sometimes a bit hard on performance, so I would recommend not using too many of them in a space. 

<img width="1427" alt="Screen Shot 2020-07-27 at 1 53 00 PM" src="https://user-images.githubusercontent.com/6835324/88591004-87864800-d010-11ea-936d-e341830be81a.png">

**Comment by j-conrad** [comment, marked as answer]
Glass and transparent surfaces are some of the trickier things to make work-- not just in Hubs but with the threejs renderer in general (which Hubs uses).
As mentioned in a previous answer, the Architecture Kit in Spoke has some semi-transparent window pieces that use an alpha channel to control the opacity. They are set to 'Alpha Blend' in their material settings. While this works, it's a bit more expensive when running on mobile or lower-powered graphics processors. Also, there are some visual side-effects that may occur when multiple transparent surfaces overlap.
If you use transparency at all, it's best to stick with 'Alpha Clip' mode. It avoids the common side-effects with overlapping transparent surfaces and is much cheaper to render. However, it wont' give you semi-transparency-- only binary "fully transparent or fully opaque" parts.
It's often simpler to just make windows non-existent.
For other things--like maybe a glass bottle or something more three dimensional-- you can use 'Alpha Blend' but be aware of the possible visual side-effects.

**Reply by Thitle** [reply]
I did tests using blender for glass in Alpha blend but when I export to spoke it doesn't really work, would you have any configuration steps for blender?

**Comment by Thitle** [comment]
Thank you, I will test and send news
